server/src/main/java/org/jboss/as/server/deployment/module/ModuleExtensionListProcessor.java

```diff
@@ -83,12 +83,12 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
             }
         }
 
-        for (ModuleIdentifier extension : allExtensionListEntries) {
-            ModuleDependency dependency = ModuleDependency.Builder.of(moduleLoader, extension.toString()).setImportServices(true).setUserSpecified(true).build();
+        for (String extension : allExtensionListEntries) {
+            ModuleDependency dependency = ModuleDependency.Builder.of(moduleLoader, extension).setImportServices(true).setUserSpecified(true).build();
             dependency.addImportFilter(PathFilters.getMetaInfSubdirectoriesFilter(), true);
             dependency.addImportFilter(PathFilters.getMetaInfFilter(), true);
             moduleSpecification.addLocalDependency(dependency);
-            nextPhaseDeps.add(ServiceModuleLoader.moduleSpecServiceName(extension.toString()));
+            nextPhaseDeps.add(ServiceModuleLoader.moduleSpecServiceName(JBossModulesNameUtil.parseCanonicalModuleIdentifier(extension)));
```

Comment:
Note to the reviewer:

I think parsing the extension here to ensure a canonicalized name is optional, at this stage, the "extension" string should be something without a slot, since it comes from Attachments.EXTENSION_LIST_ENTRIES which is populated at https://github.com/wildfly/wildfly-core/blob/035d41260346840f0073a90875997f2b8f16df17/server/src/main/java/org/jboss/as/server/deployment/module/ManifestExtensionListProcessor.java#L76-L77

So, I am not sure this parse here will never do anything. I left it because it sounds more robust, but I think we can also remove it.

Looks like a minor detail though, so if there is no objection, I'm fine going as it is.

Reply:
My preference is to remove it. I'd prefer we be rigorous about canonicalizing as close to the questionable external input as possible, javadoc as requiring canonical names any APIs that are called from the class that canonicalized external input, and otherwise regard names as canonical. 

We can end up recanonicalizing all over the place. It's not a super-cheap operation.